deepchecks/metric_utils.py

```diff
@@ -64,12 +64,11 @@ def task_type_check(model: Union[ClassifierMixin, RegressorMixin], dataset: 'Dat
     model_type_validation(model)
     dataset.validate_label(task_type_check.__name__)
 
-    if getattr(model, 'predict_proba', None):
+    if hasattr(model, 'predict_proba'):
         unique_labels = dataset.label_col().unique()
         if sorted(unique_labels) != list(range(len(unique_labels))):
             raise DeepchecksValueError(f'Classification labels must be a consecutive set from 0 to MAX_LABEL,'
                                        f' found {sorted(unique_labels)}.')
-        model: ClassifierMixin
```

Comment:
this used to serve to help my type hint in the IDE

Reply:
hm/
1. how is it helping if `model` var is not used below that annotation?
2. I am using VSCode with the pyright static type checker, and that is what he is saying if that annotation is present - `Parameter declaration "model" is obscured by a declaration of the same name`
3. for cases like this it is better to use `typing.cast` (or `assert`), mypy docs about it - [link](https://mypy.readthedocs.io/en/stable/casts.html), python docs about it - [link](https://docs.python.org/3/library/typing.html#typing.cast)